tf2onnx/onnx_opset/tensor.py

```diff
@@ -2043,6 +2043,337 @@ def version_11(cls, ctx, node, **kwargs):
+            expanded_diag_graph.add_graph_output(expanded_diag.output[0], ctx.get_dtype(diag), expanded_diag_shape)
+            return expanded_diag_graph
+
+        # if shape of diag is [-1], expand it to [1, -1]
+        if_expanded_diag = ctx.make_node("If", [initial_diag_rank_greater.output[0]])
+        if_expanded_diag.set_body_graph_as_attr("then_branch", IdentityDiagGraph())
+        if_expanded_diag.set_body_graph_as_attr("else_branch", ExpandDiagGraph())  # processed diag
+        temp_diag_shape = ctx.make_node("Shape", [if_expanded_diag.output[0]])
+        temp_diag_depth = ctx.make_node("Slice",
+                                        [temp_diag_shape.output[0], const_neg_two.output[0], const_neg_one.output[0]])
+        depth_match = ctx.make_node("Equal", [k_count.output[0], temp_diag_depth.output[0]])
+
+        def IdentityExpanedGraph():
+            identity_expanded_diag_graph = ctx.create_new_graph_with_same_config()
+            identity_expanded_diag_graph.parent_graph = ctx
+            identity_expanded_diag = identity_expanded_diag_graph.make_node("Identity", [if_expanded_diag.output[0]])
+            identity_expanded_diag_graph.add_graph_output(identity_expanded_diag.output[0], ctx.get_dtype(diag),
+                                                          expanded_diag_shape)
+            return identity_expanded_diag_graph
+
+        def FurtherExpanedDiag():
+            further_expanded_diag_graph = ctx.create_new_graph_with_same_config()
+            further_expanded_diag_graph.parent_graph = ctx
+            further_expanded_diag = further_expanded_diag_graph.make_node("Unsqueeze", [if_expanded_diag.output[0]],
+                                                                          attr={'axes': [-2]})
+            further_expanded_diag_graph.add_graph_output(further_expanded_diag.output[0], ctx.get_dtype(diag),
+                                                         expanded_diag_shape)
+            return further_expanded_diag_graph
+
+        # if depth of diag is not equal to k_max - k_min + 1, which is actually 1, expand diag on axis -2
+        final_expanded_diag = ctx.make_node("If", [depth_match.output[0]])
+        final_expanded_diag.set_body_graph_as_attr("then_branch", IdentityExpanedGraph())
+        final_expanded_diag.set_body_graph_as_attr("else_branch", FurtherExpanedDiag())
+        # compute diag_len which is the length of diagonal when k is 0
+        diag_shape = ctx.make_node("Shape", [final_expanded_diag.output[0]])
+        half_diag_shape = ctx.make_node("Slice", [diag_shape.output[0], const_zero.output[0], const_neg_two.output[0]])
+        diag_rank = ctx.make_node("Shape", [diag_shape.output[0]])
+        diag_width = ctx.make_node("Slice", [diag_shape.output[0], const_neg_one.output[0], diag_rank.output[0]])
+        k_range = ctx.make_node("Range", [k_min.output[0], k_max_inc.output[0]], domain="com.microsoft")
```

Comment:
Should not use domain="com.microsoft" anymore -- since Range is an official function now.